query/src/interpreters/interpreter_select_v2.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_exception::Result;
+use common_streams::SendableDataBlockStream;
+use common_tracing::tracing;
+
+use crate::interpreters::stream::ProcessorExecutorStream;
+use crate::interpreters::Interpreter;
+use crate::interpreters::InterpreterPtr;
+use crate::pipelines::new::executor::PipelinePullingExecutor;
+use crate::sessions::QueryContext;
+use crate::sql::Planner;
+
+/// Interpret SQL query with new SQL planner
+pub struct SelectInterpreterV2 {
+    ctx: Arc<QueryContext>,
+    query: String,
+}
+
+impl SelectInterpreterV2 {
```

Comment:
Maybe we can directly use the current `SelectInterpreter`

Reply:
Seems we can add a new trait like `execute2` in SelectInterpreter for the new planner, this what new processor did.

Reply:
And `InterpreterFactory` should add a new factory method like `get2` too.